metricbeat/module/kubernetes/state_replicaset/state_replicaset.go

```diff
@@ -91,18 +87,43 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	}, nil
 }
 
-// Fetch methods implements the data gathering and data conversion to the right format
-// It returns the event which is then forward to the output. In case of an error, a
-// descriptive error must be returned.
-func (m *MetricSet) Fetch() ([]common.MapStr, error) {
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
 	m.enricher.Start()
 
 	events, err := m.prometheus.GetProcessedMetrics(mapping)
-	if err == nil {
-		m.enricher.Enrich(events)
+	if err != nil {
+		m.Logger().Error(err)
+		reporter.Error(err)
+		return
+	}
+
+	m.enricher.Enrich(events)
+	for _, event := range events {
+
+		var moduleFieldsMapStr common.MapStr
+		moduleFields, ok := event[mb.ModuleDataKey]
+		if ok {
```

Comment:
Can you file a follow up issue so we improve the prometheus input so we can get rid of this. I think @odacremolbap worked already on some parts.

Reply:
Sure, I did some quick test using this https://github.com/elastic/beats/blob/master/metricbeat/module/kubernetes/state_container/state_container_test.go#L29 but it didn't work :(

Reply:
I didn't understand why some metricsets were using it and some not

Reply:
Not sure I can follow your above comment. I'm ok with the current hack but want to make sure we follow up that is why the ask for an issue.

Reply:
As soon as we have migrated all the new interface, I think we can remove the old testing methods. I wanted to keep them around for the migration to make sure we don't miss things.

Reply:
Issue opened https://github.com/elastic/beats/issues/11624